lib/edda-server/src/change_set_processor_task/materialized_view.rs

```diff
@@ -754,356 +763,249 @@ where
 /// The [`Change`] of the trigger entity that would have spawned a build task for the
 /// `mv_kind`, if we hadn't already reached the concurrency limit for running build
 /// tasks.
-#[derive(Debug)]
+#[derive(Debug, Eq, PartialEq)]
 struct QueuedBuildMvTask {
     pub change: Change,
     pub mv_kind: ReferenceKind,
+    pub priority: BuildPriority,
+}
+
+impl Ord for QueuedBuildMvTask {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match self.priority.cmp(&other.priority) {
+            std::cmp::Ordering::Equal => {
+                // Within the same priority, items will be ordered by the triggering
+                // entity ID.
+                self.change.entity_id.cmp(&other.change.entity_id)
+            }
+            ord => ord,
```

Comment:
The sorting here is actually backwards. This should have "higher priority" items as "larger".